src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -846,6 +846,72 @@ public void VectorSingleEqualsNaNTest()
             var nan = new Vector<float>(float.NaN);
             Assert.True(nan.Equals(nan));
         }
+
+        [Fact]
+        public void VectorDoubleEqualsNonCanonicalNaNTest()
+        {
+            var cast = BitConverter.DoubleToUInt64Bits(double.NaN);
+            var one = BitConverter.DoubleToUInt64Bits(1);
+            var sign = one ^ BitConverter.DoubleToUInt64Bits(-1);
```

Comment:
This gets you just the sign much more simply:
```suggestion
            var sign = BitConverter.DoubleToUInt64Bits(-0.0);
```

Reply:
Also could just do `0x8000_0000_0000_0000`

Reply:
using .CopySign as suggested below